RNS Guide with the New SDK (#444)
rsksmart/devportal: docs/02-developers/06-integrate/03-rns/js-sdk.md

sidebar_label: RNS Javascript SDK
sidebar_position: 300
title: RIF Name Service (RNS) Javascript SDK
description: RNS Javascript SDK for domain registration, management, and resolution
tags: [rns, integrate, integration guide, rif, sdk]
---

> The `@rsksmart/rns-sdk` package helps you **interact with the Rootstock Name Service (RNS)**. It lets you:

Reviewer: ```suggestion
The `[@rsksmart/rns-sdk](https://github.com/rsksmart/rns-sdk)` package helps you **interact with the Rootstock Name Service (RNS)**. You can:
```

- Register .rsk domains
- Check domain and subdomain availability
- Query and set domain ownership
- Resolve domains to addresses
- Manage subdomains
- Work with partner registrars

## Installation

Install the package using npm:

```bash
npm install @rsksmart/rns-sdk ethers
```

## Contract Addresses

You will need these addresses to initialize the SDK:

> | Contract | RSK Mainnet | RSK Testnet |

Reviewer: Rootstock Testnet | Mainnet

|----------|-------------|-------------|
| RNS Registry | `0xcb868aeabd31e2b66f74e9a55cf064abb31a4ad5` | `0x7d284aaac6e925aad802a53c0c69efe3764597b8` |
| RIF Token | `0x2acc95758f8b5f583470ba265eb685a8f45fc9d5` | `0x19f64674d8a5b4e652319f5e239efd3bc969a1fe` |
| RSK Owner | `0x45d3e4fb311982a06ba52359d44cb4f5980e0ef1` | `0xca0a477e19bac7e0e172ccfd2e3c28a7200bdb71` |
| FIFS Addr Registrar | `0xd9c79ced86ecf49f5e4a973594634c83197c35ab` | `0x90734bd6bf96250a7b262e2bc34284b0d47c1e8d` |
| Partner Registrar | - | `0x8104d97f6d82a7d3afbf45f72118fad51f190c42` |

## SDK Classes

The SDK provides five main classes:

| Class | Purpose |
|-------|---------|
| `RNS` | Domain management (owner, resolver, subdomains) |
| `AddrResolver` | Address resolution |
| `RSKRegistrar` | Standard .rsk domain registration |
| `PartnerRegistrar` | Partner-based domain registration |
| `PartnerConfiguration` | Partner contract configuration |

---

## 1. RNS Class

The `RNS` class handles domain management operations.

> ### Constructor

Reviewer: Kindly inform the user at what point the initialization of the SDK happens after adding the contract addresses.

```js
import { Signer } from 'ethers'
import { RNS } from '@rsksmart/rns-sdk'

let signer: Signer
const rns = new RNS(registryAddress, signer)
```

### Methods

#### 1. `getOwner(domain)`

Gets the controller/owner of a domain.

```js
const owner = await rns.getOwner('mydomain.rsk')
console.log('Owner:', owner)
```

#### 2. `setOwner(domain, newController)`

Transfers domain ownership to a new controller.

```js
const domain = 'mydomain.rsk'
const newController = '0xb774...d771'

const tx = await rns.setOwner(domain, newController)
await tx.wait()
console.log('Ownership transferred!')
```

#### 3. `resolver(domain)`

> Gets the resolver contract address for a domain.

Reviewer: For each method, link to the specific section in the [README](https://github.com/rsksmart/rns-sdk?tab=readme-ov-file#resolver), for future reference and easy maintainability
